state/workers.go

```diff
@@ -48,9 +50,15 @@ func newWorkers(st *State) (*workers, error) {
 			Clock:        st.clock(),
 		}),
 	}
-	ws.StartWorker(txnLogWorker, func() (worker.Worker, error) {
-		return watcher.New(st.getTxnLogCollection()), nil
-	})
+	if hub == nil {
```

Comment:
would it be better to be passing a func()=>hub rather than the raw hub ?
I'm just wondering about lifetime of various objects and the chance that something gets restarted.

Reply:
The hub is owned by the state pool, so it will always outlast any state object in the pool.